sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -1048,6 +1046,30 @@ export class MessageReceiver extends LinkEntity {
         delivery.reject(options.error || {});
       }
     });
+
+    const config: RetryConfig<any> = {
+      operation: () => {
+        return settleMessagePromise;
+      },
+      connectionId: this._context.namespace.connectionId,
+      operationType: RetryOperationType.management,
+      delayInSeconds: Constants.defaultDelayBetweenRetriesInSeconds,
+      times: Constants.defaultRetryAttempts
+    };
+    try {
+      return retry<Promise<any>>(config);
+    } catch (err) {
+      this._deliveryDispositionMap.delete(message.delivery.id);
```

Comment:
This is already taken care of in the code paths where the promise returned by `settleMessagePromise` is rejected. Why do we need to catch the error and delete the map again?

Reply:
The map is cleared only from the onSettled handler. It's possible that the onSettled handler never gets invoked and so the timeout error code path to permanent rejection would clear it.

Reply:
In case of non timeout error scenarios, we will now be running the delete call twice.
A cleaner solution would be to move this to `finally` and remove it from other places

Reply:
Updated

Reply:
This approach doesn't work and were causing tests to fail, so had to revert it